compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -177,3 +178,10 @@ internal fun KtClassOrObject.generateClassName(
             .shortName()
             .asString()
       }
+
+internal fun List<KotlinType>.getAllSuperTypes(): Sequence<FqName> =
+  generateSequence(this) { kotlinTypes ->
+    if (kotlinTypes.isEmpty()) null else kotlinTypes.flatMap { it.supertypes() }
```

Comment:
Super minor nit: `kotlinTypes.takeUnless { it.isEmpty() }?.flatMap { it.supertypes() }`

Reply:
I think that's even easier to read `kotlinTypes.ifEmpty { null }?.flatMap { it.supertypes() }`.